pkg/networks/commands.go

```diff
@@ -23,22 +28,60 @@ func (config *NetworksConfig) Check(name string) error {
 	return fmt.Errorf("network %q is not defined", name)
 }
 
+// IsDaemonInstalled checks whether the daemon is installed.
+func (config *NetworksConfig) IsDaemonInstalled(daemon string) (bool, error) {
+	var p string
+	switch daemon {
+	case VDESwitch:
+		p = config.Paths.VDESwitch
+	case VDEVMNet:
+		p = config.Paths.VDEVMNet
+	case SocketVMNet:
+		p = config.Paths.SocketVMNet
+	default:
+		return false, fmt.Errorf("unknown daemon type %q", daemon)
+	}
+	if p == "" {
+		return false, nil
+	}
+	if _, err := exec.LookPath(p); err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// Sock returns a socket_vmnet socket.
+func (config *NetworksConfig) Sock(name string) string {
+	return fmt.Sprintf("%s/socket_vmnet.%s", config.Paths.VarRun, name)
+}
+
+// VDESock returns a vde socket.
+//
+// Deprecated. Use Sock.
 func (config *NetworksConfig) VDESock(name string) string {
 	return filepath.Join(config.Paths.VarRun, fmt.Sprintf("%s.ctl", name))
 }
 
 func (config *NetworksConfig) PIDFile(name, daemon string) string {
-	return filepath.Join(config.Paths.VarRun, fmt.Sprintf("%s_%s.pid", name, daemon))
+	daemonTrimmed := strings.TrimPrefix(daemon, "vde_") // for compatibility
+	return filepath.Join(config.Paths.VarRun, fmt.Sprintf("%s_%s.pid", name, daemonTrimmed))
 }
 
 func (config *NetworksConfig) LogFile(name, daemon, stream string) string {
 	networksDir, _ := dirnames.LimaNetworksDir()
-	return filepath.Join(networksDir, fmt.Sprintf("%s_%s.%s.log", name, daemon, stream))
+	daemonTrimmed := strings.TrimPrefix(daemon, "vde_") // for compatibility
+	return filepath.Join(networksDir, fmt.Sprintf("%s_%s.%s.log", name, daemonTrimmed, stream))
 }
 
 func (config *NetworksConfig) User(daemon string) (osutil.User, error) {
+	if ok, _ := config.IsDaemonInstalled(daemon); !ok {
+		return osutil.User{}, fmt.Errorf("daemon %q is not available", daemon)
+	}
```

Comment:
I'm not sure why this condition is needed. I think it is benign, but why should `User()` return an error when the daemon is not installed? This will be caught before the daemon is started, won't it?

Reply:
Further testing shows that `startDaemon` will call `StartDaemon` unless the `User` call fails, which sounds a weird way to control this, but I guess that means the check is is required for now.

However, the error message is not really informative:

```
INFO[0004] Starting vde_switch daemon for "shared" network
FATA[0004] daemon "vde_switch" is not available
```

It doesn't tell me where it expects the daemon to be installed.

Reply:
Updated to print the daemon path in the error